### PR TITLE
http-tap downstream connection

### PR DESCRIPTION
**Pull request by durd07**
Commit Message: trace downstream connection in http-tap
Additional Description:
Risk Level: Low
Testing: Ut/Integrated Test
Docs Changes: 
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by durd07** [comment]
@lizan Could you please help to review this?
Thanks in advance.

**Comment by durd07** [comment]
@envoyproxy/senior-maintainers 
Could you please help to review this? As @lizan is not around for a long time.

File: source/extensions/filters/http/tap/tap_filter.cc [hunk 1]
```diff
@@ -69,6 +69,13 @@ Http::FilterTrailersStatus Filter::encodeTrailers(Http::ResponseTrailerMap& trai
   return Http::FilterTrailersStatus::Continue;
 }
 
+void Filter::setDownstreamConnectionAddress(
+    const Envoy::Network::ConnectionInfoProvider& connection_info_provider) {
+  if (tapper_ != nullptr) {
+    tapper_->setDownstreamConnectionAddress(connection_info_provider);
+  }
+}
+
```
Comment (lizan) [inline source/extensions/filters/http/tap/tap_filter.cc:78]: You shouldn't need all these from StreamFilterBase to here. The downstream connection is available via StreamFilterCallbacks::connection, or you can get them in the log method below from StreamInfo.

File: api/envoy/data/tap/v3/common.proto [hunk 2]
```diff
@@ -36,3 +38,14 @@ message Body {
   // <envoy_v3_api_field_config.tap.v3.OutputConfig.max_buffered_tx_bytes>` settings.
   bool truncated = 3;
 }
+
+// Connection properties.
+message Connection {
```
Comment (lizan) [inline api/envoy/data/tap/v3/common.proto:43]: just to double check, moving this doesn't break generated code compatibility? cc @envoyproxy/api-shepherds

**Review by lizan — CHANGES_REQUESTED** [review]

File: api/envoy/data/tap/v3/common.proto (hunk 2 above)
Comment (adisuissa) [reply api/envoy/data/tap/v3/common.proto:43]: I think that should be ok, as the package name is the same.

File: source/extensions/filters/http/tap/tap_filter.cc (hunk 1 above)
Comment (durd07) [reply source/extensions/filters/http/tap/tap_filter.cc]: @lizan Thanks, you're right. I have changed code to get the connection from StreamFilterCallbacks::connection, please help to review this again.

**Comment by durd07** [comment]
@lizan Please help to review again.

**Review by lizan — APPROVED** [review]

**Comment by durd07** [comment]
@mattklein123 Could you please help to review and merge this PR?
Thanks in advance.

**Review by lizan — APPROVED** [review]

**Review by lizan — APPROVED** [review]

**Review by lizan — APPROVED** [review]

**Pull request merged**